torch_em/data/datasets/bcss.py

```diff
@@ -0,0 +1,128 @@
+            15: undetermined; 16: nerve; 17: skin_adnexa; 18: blood_vessel; 19: angioinvasion; 20: dcis; 21: other
+        """
+        self.label_choices = label_choices
+
+    def __call__(
+            self,
+            labels: np.ndarray
+    ) -> np.ndarray:
+        """Returns the transformed labels
+        """
+        if self.label_choices is not None:
+            _segmentation = np.zeros(labels.shape)
+            for uq_label in self.label_choices:
+                _segmentation[labels == uq_label] = labels[labels == uq_label]
+            segmentation, _, _ = vigra.analysis.relabelConsecutive(_segmentation.astype("uint64"))
+        else:
+            segmentation, _, _ = vigra.analysis.relabelConsecutive(labels)
+
+        return segmentation
+
+
+def _download_bcss_dataset(path, download):
+    """Current recommendation:
+        - download the folder from URL manually
+        - use the consortium's git repo to download the dataset (https://github.com/PathologyDataScience/BCSS)
+    """
+    raise NotImplementedError("Please download the dataset using the drive link / git repo directly")
+
+    # FIXME: limitation for the installation below:
+    #   - only downloads first 50 files - due to `gdown`'s download folder function
+    #   - (optional) clone their git repo to download their data
+    util.download_source_gdrive(path=path, url=URL, download=download, checksum=CHECKSUM, download_type="folder")
+
+
+def get_bcss_dataset(path, patch_shape, label_choices, download=False, label_dtype=torch.int64, **kwargs):
+    """Dataset for breast cancer tissue segmentation in histopathology.
+
+    This dataset is from https://bcsegmentation.grand-challenge.org/BCSS/.
+    Please cite this paper (https://doi.org/10.1093/bioinformatics/btz083) if you use this dataset for a publication.
+    """
+    if download:
+        _download_bcss_dataset(path, download)
+
+    # when downloading the files from `URL`, the input images are stored under `rgbs_colorNormalized`
+    # when getting the files from the git repo's command line feature, the input images are stored under `images`
+    if os.path.exists(os.path.join(path, "images")):
+        image_paths = sorted(glob(os.path.join(path, "images", "*")))
+        label_paths = sorted(glob(os.path.join(path, "masks", "*")))
+    elif os.path.exists(os.path.join(path, "0_Public-data-Amgad2019_0.25MPP", "rgbs_colorNormalized")):
+        image_paths = sorted(glob(os.path.join(path, "0_Public-data-Amgad2019_0.25MPP", "rgbs_colorNormalized", "*")))
+        label_paths = sorted(glob(os.path.join(path, "0_Public-data-Amgad2019_0.25MPP", "masks", "*")))
+    else:
+        raise ValueError(
+            "Please check the image directory. If downloaded from gdrive, it's named \"rgbs_colorNormalized\", if from github it's named \"images\""
+        )
+    assert len(image_paths) == len(label_paths)
+
+    label_trafo = BCSSLabelTrafo(label_choices=label_choices)
```

Comment:
This will lead to problems if the user gives a label trafo.

Reply:
Okay. Let's discuss this tomorrow what's the best strategy to take care of this.